scripts/utils/migrate/taxonomy-redirects.js

```diff
@@ -29,24 +29,42 @@ const getTaxonomyPath = async (uri) => {
   }
 };
 
+const getNodePath = async (uri) => {
+  const id = uri.replace(/.*\/node\//, '');
+  const url = new URL(`/api/migration/content/node/${id}/urlPath`, BASE_URL);
+  const headers = {
+    'Content-Type': 'application/json',
+    'X-Phpshield-Key-Disable': process.env.ACQUIA_DEV_PHP_SHIELD_KEY,
+  };
+  try {
+    const res = await fetch(url, { headers });
+    const json = await res.json();
+    return get(json, 'docs[0].doc.path');
+  } catch (error) {
+    return null;
+  }
+};
+
 const fetchTaxonomyRedirects = async (redirects) => {
   const taxonomy = await Promise.all(
     Object.entries(redirects)
       .filter(([uri]) => uri.startsWith('/taxonomy'))
       .map(async ([url, paths]) => [await getTaxonomyPath(url), paths])
   );
 
-  return taxonomy.reduce((memo, [url, paths]) => {
-    const filteredPaths = paths.filter(
-      (path) =>
-        path !== url &&
-        !path.startsWith('/node') &&
-        !path.startsWith('/taxonomy')
+  return await taxonomy.reduce(async (memo, [url, paths]) => {
```

Comment:
Let's try and avoid reducing on an `async` function. They are a bit hard to manage. That `await taxonomy.reduce` isn't doing what you think its doing 😄 

This `taxonomy.reduce` is actually going to return an array of promises. Unfortunately reducing on an array of promises doesn't resolve when all of the items have been resolved. As a test, try this:

```js
const reducePromises = async () => {
  const nums = [1, 2, 3]
  return await nums.reduce(async (memo, num) => {
     return memo + num
  }, 0)
}

const test = async () => {
  const result = await reducePromises()  
  
  console.log(result)
}

test()
```

What you'll get is `[object Object]3`, not quite what you'd expect. Instead, I find it best to break this up into 2 steps:

1) do all the async work in a `Promise.all`
2) Reduce on that value

Since you're trying to fetch data in here, I'd break it up that way:

```
const filteredPaths = await Promise.all(
  taxonomy
    .filter((path => ...)
    .map(async ([url, paths]) => ...)
)

return filteredPaths.reduce(...)
```

I think you'll find this to be much more reliable